tests/e2e/e2e_test.go

```diff
@@ -41,67 +42,98 @@ func (s *IntegrationTestSuite) Test01IBCTokenTransfer() {
 // This is the reason for prefixing the name with 02 to ensure
 // correct order.
 func (s *IntegrationTestSuite) Test02CreatePool() {
-	chain := s.configurer.GetChainConfig(0)
-	node, err := chain.GetDefaultNode()
+	chainA := s.configurer.GetChainConfig(0)
+	chainB := s.configurer.GetChainConfig(1)
+	chainANode, err := chainA.GetDefaultNode()
+	s.NoError(err)
+	chainBNode, err := chainB.GetDefaultNode()
 	s.NoError(err)
 
-	node.CreatePool("nativeDenomPool.json", initialization.ValidatorWalletName)
+	chainANode.CreatePool("nativeDenomPool.json", initialization.ValidatorWalletName)
+	chainBNode.CreatePool("nativeDenomPool.json", initialization.ValidatorWalletName)
 
 	if s.skipIBC {
 		s.T().Log("skipping creating pool with IBC denoms because IBC tests are disabled")
 		return
 	}
 
-	node.CreatePool("ibcDenomPool.json", initialization.ValidatorWalletName)
+	chainANode.CreatePool("ibcDenomPool.json", initialization.ValidatorWalletName)
+	chainBNode.CreatePool("ibcDenomPool.json", initialization.ValidatorWalletName)
 }
 
-// Test03SuperfluidVoting tests that superfluid voting is functioning as expected.
+// Test03AddToExistingLockPostUpgrade tests lockups to both regular and superfluid locks.
+// Specifically, we ensure addToExistingLock works for both preupgrade and postupgrade locks.
+// This must be run after Test02CreatePool, otherwise, in the event we skip upgrade testing
+// we will not have any gamm pool assets to lock up.
+// This is the reason for prefixing the name with 03 to ensure
+// correct order.
+func (s *IntegrationTestSuite) Test03AddToExistingLockPostUpgrade() {
+	chainA := s.configurer.GetChainConfig(0)
+	chainB := s.configurer.GetChainConfig(1)
+	chainANode, err := chainA.GetDefaultNode()
+	s.NoError(err)
+	chainBNode, err := chainB.GetDefaultNode()
+	s.NoError(err)
+	if s.skipUpgrade {
+		// enable superfluid assets
+		chainA.EnableSuperfluidAsset("gamm/pool/1")
+		chainB.EnableSuperfluidAsset("gamm/pool/1")
```

Comment:
So it seems that the superfluid tests depend on this when the upgrade is disabled. Can we make each test independent by making them operate on different assets, pools etc?

The goal is to be able to run each test in isolation

Reply:
https://github.com/osmosis-labs/osmosis/issues/2569